### PR TITLE
HW12 Andrii Voinenko

### PR DESCRIPTION
**Pull request by andrii-voinenko**

### CODE REVIEW

File: Andrii_Voinenko/Lesson12/task1.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
+class Sony:
+    def __init__(self, department: str, location: str, company_name='Sony'):
+        self.__department = department
+        self.__location = location
+        self.__company_name = company_name
+
+        if self.__company_name != ('Sony' or 'sony'):
+            raise TypeError('Only "Sony" company name available')
```
Comment (bohdandovbsyh) [inline Andrii_Voinenko/Lesson12/task1.py]: why only sony? Tha task was about the genera company class. Please fix it

File: Andrii_Voinenko/Lesson12/task1.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        """
+        Allows to set company location if it in company_locations list
+        """
+        company_locations = ['China', 'EU', 'US']
+
+        if new_location in company_locations:
+            self.__location = new_location
+        else:
+            raise Warning('You cannot set not approved location')
+
```
Comment (bohdandovbsyh) [inline Andrii_Voinenko/Lesson12/task1.py:89]: also, add some functions. not only @property and setters

File: Andrii_Voinenko/Lesson12/task2.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        Return employee work position
+        """
+        return self.__position
+
+    @name.setter
+    def name(self, new_name):
+        """
+        Allows set new name to employee
+        """
+        self.__name = new_name
```
Comment (bohdandovbsyh) [inline Andrii_Voinenko/Lesson12/task2.py]: no validation. I can set empty name here

File: Andrii_Voinenko/Lesson12/task2.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+    @salary.setter
+    def salary(self, new_salary):
+        """
+        Allows set new salary to employee
+        """
+        self.__salary = new_salary
```
Comment (bohdandovbsyh) [inline Andrii_Voinenko/Lesson12/task2.py]: new salary can be any number. So it can be case when employee pays money to company

**Comment by andrii-voinenko** [comment]
Added fixes as was mentioned in comments

**Review by bohdandovbsyh — APPROVED** [review]

**Pull request merged**